src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3504,6 +3505,56 @@ private BoundExpression BindNullCoalescingOperator(BinaryExpressionSyntax node,
             return GenerateNullCoalescingBadBinaryOpsError(node, leftOperand, rightOperand, Conversion.NoConversion, diagnostics);
         }
 
+
+        private BoundExpression BindNullCoalescingAssignmentOperator(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            BoundExpression leftOperand = BindValue(node.Left, diagnostics, BindValueKind.CompoundAssignment);
+            BoundExpression rightOperand = BindValue(node.Right, diagnostics, BindValueKind.RValue);
+
+            // If either operand is bad, bail out preventing more cascading errors
+            if (leftOperand.HasAnyErrors || rightOperand.HasAnyErrors)
+            {
+                return new BoundNullCoalesingAssignmentOperator(node, leftOperand, rightOperand, CreateErrorType(), hasErrors: true);
+            }
+
+            // Unlike a standard null coalescing expression, the resulting type of the expression a ??= b
+            // must be A (where A is the type of a), as it is where the result of the assignment will end up. Therefore,
+            // we must ensure that B (where B is the type B) is implicitly convertible to A or A0 (where A0 is the
+            // underlying non-nullable type of A).
+            TypeSymbol leftType = leftOperand.Type;
+            Debug.Assert(leftType != null);
+            TypeSymbol optRightType = rightOperand.Type;
+            bool rightTypeIsNonNull = !(optRightType is null);
+            bool isLeftNullable = leftType.IsNullableType();
+
+            // If A exists and is not a nullable type or reference type, a compile-time error occurs
```

Comment:
>A exists [](start = 18, length = 8)

We're not checking that here. Consider removing from comment here and comment below.